File: python/commands/valid_commands/amplifier_supply.py
from commands.valid_commands.base import CommandBase


class AmplifierSupplyCmd(CommandBase):
    TAG: str = '0x0'
    NAME: str = 'питание усилителя'
    _allowed_values: list = ['0', '1']

    def validate(self):
        return self.value in self._allowed_values

File: rust/commands/valid_commands/amplifier_supply.rs
const TAG: &str = "0x0";
const NAME: &str = "питание усилителя";
const ALLOWED_VALUES: [&str; 2] = ["0", "1"];

pub struct AmplifierSupplyCmd {
    pub value: String,
}

impl AmplifierSupplyCmd {
    pub fn new(value: String) -> Self {
        AmplifierSupplyCmd { value }
    }

    pub fn validate(&self) -> bool {
        ALLOWED_VALUES.contains(&self.value.as_str())
    }

    pub fn name(&self) -> &str {
        NAME
    }

    pub fn tag(&self) -> &str {
        TAG
    }
}